tasks/grafana.py

```diff
@@ -127,9 +128,171 @@ def create_dashboard_in_prod(folder_name, **context):
 
 def get_available_folders():
     try:
-        response = requests.get(f"{PROD_HOST}/api/folders", headers=HEADERS_PROD, verify=False)
+        response = requests.get(f"{TEST_HOST}/api/folders", headers=HEADERS_TEST, verify=False)
         response.raise_for_status()
         folders = response.json()
         return [folder["title"] for folder in folders]
     except requests.exceptions.RequestException as e:
         raise AirflowException(f"Error fetching available folders: {str(e)}")
+
+def get_all_dashboards():    
+    try:
+        response = requests.get(f"{TEST_HOST}/api/search?type=dash-db", headers=HEADERS_PROD, verify=False)
+        response.raise_for_status()
+        return response.json()
+    except requests.exceptions.RequestException as e:
+        raise AirflowException(f"Error fetching all dashboards: {str(e)}")
+
+def get_dashboard_versions(dashboard_uid, host, headers):
+    try:
+        response = requests.get(
+            f"{host}/api/dashboards/uid/{dashboard_uid}/versions",
+            headers=headers,
+            verify=False
+        )
+        response.raise_for_status()
+        return response.json()
+    except requests.exceptions.RequestException as e:
+        raise AirflowException(f"Error fetching dashboard versions: {str(e)}")
+
+def get_latest_version(dashboard_uid, host, headers):
+    versions = get_dashboard_versions(dashboard_uid, host, headers)
+    return versions[0] if versions else None
+
+def get_stored_versions(backup_dir):
+    versions = {}
+    if os.path.exists(backup_dir):
+        for file in os.listdir(backup_dir):
+            if file.endswith('.json'):
```

Comment:
If you want to avoid to fall into the indenting rabithall, it's better to inverse the logic
```suggestion
            if not file.endswith('.json'):
              continue
            try:
              with open(..)
```